File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        romInt = {
            "M":1000,
            "D":500,
            "C":100,
            "L":50,
            "X":10,
            "V":5,
            "I":1,
        }

        ans = 0
        prev = 0
        for a in s[::-1]:
            if prev > romInt[a]:
                ans -= romInt[a]
            else:
                ans += romInt[a]
            prev = romInt[a]
        return ans

File: 0013-roman-to-integer/0013-roman-to-integer.java
import java.util.*;

class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> romInt = new HashMap<>();

        romInt.put('M', 1000);
        romInt.put('D', 500);
        romInt.put('C', 100);
        romInt.put('L', 50);
        romInt.put('X', 10);
        romInt.put('V', 5);
        romInt.put('I', 1);

        int ans = 0;
        int prev = 0;

        for (int i = s.length() - 1; i >= 0; i--) {
            char a = s.charAt(i);
            if (prev > romInt.get(a)) {
                ans -= romInt.get(a);
            } else {
                ans += romInt.get(a);
            }
            prev = romInt.get(a);
        }
        return ans;
    }
}